nixos/modules/services/matrix/mautrix-discord.nix

```diff
@@ -0,0 +1,467 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+# This module is an adaptation of the module for mautrix-meta in the same directory
+let
+  settingsFormat = pkgs.formats.yaml { };
+
+  cfg = config.services.mautrix-discord;
+
+  fullDataDir = cfg: "/var/lib/${cfg.dataDir}";
+
+  settingsFile = cfg: "${fullDataDir cfg}/config.yaml";
+  settingsFileUnsubstituted = cfg: settingsFormat.generate "mautrix-discord-config.yaml" cfg.settings;
+in
+{
+  options = {
+    services.mautrix-discord = {
+
+      package = lib.mkPackageOption pkgs "mautrix-discord" { };
+
+      enable = lib.mkEnableOption "Mautrix-Discord, a Matrix <-> Discord hybrid puppeting/relaybot bridge";
+
+      dataDir = lib.mkOption {
+        type = lib.types.str;
+        default = "mautrix-discord";
+        description = ''
+          Path to the directory with database, registration, and other data for the bridge service.
+          This path is relative to `/var/lib`, it cannot start with `../` (it cannot be outside of `/var/lib`).
+        '';
+      };
+
+      registrationFile = lib.mkOption {
+        type = lib.types.path;
+        readOnly = true;
+        description = ''
+          Path to the yaml registration file of the appservice.
+        '';
+      };
+
+      registerToSynapse = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = ''
+          Whether to add registration file to `services.matrix-synapse.settings.app_service_config_files` and
+          make Synapse wait for registration service.
+        '';
+      };
+
+      settings = lib.mkOption rec {
```

Comment:
Testing out this module more today, I noticed that this option can be simplified by removing everything except for the `bridge.encryption` & `logging` settings. `mautrix-discord` will automatically fill in these defaults if they're not specified.

Then the `defaultText` could also be removed because the `default` would no longer be computed from other options.